src/mlpack/tests/main_tests/kde_test.cpp

```diff
@@ -405,4 +405,120 @@ BOOST_AUTO_TEST_CASE(KDEMainInvalidRelativeError)
+
+  Log::Fatal.ignoreInput = true;
+  // Invalid under 0.
+  SetInputParam("mc_probability", -0.1);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Invalid over 1.
+  SetInputParam("mc_probability", 1.1);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Valid value.
+  SetInputParam("mc_probability", 0.3);
+  BOOST_REQUIRE_NO_THROW(mlpackMain());
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensure we get an exception when an invalid Monte Carlo initial sample size
+  * is specified.
+ **/
+BOOST_AUTO_TEST_CASE(KDEMainInvalidMCInitialSampleSize)
+{
+  arma::mat reference = arma::randu<arma::mat>(1, 10);
+  arma::mat query = arma::randu<arma::mat>(1, 5);
+
+  // Main params.
+  SetInputParam("reference", reference);
+  SetInputParam("query", query);
+  SetInputParam("kernel", std::string("gaussian"));
+  SetInputParam("monte_carlo", true);
+
+  Log::Fatal.ignoreInput = true;
+  // Invalid under 0.
+  SetInputParam("initial_sample_size", -1);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Invalid 0.
+  SetInputParam("initial_sample_size", 0);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Valid value.
+  SetInputParam("initial_sample_size", 20);
+  BOOST_REQUIRE_NO_THROW(mlpackMain());
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensure we get an exception when an invalid Monte Carlo entry coefficient
+  * is specified.
+ **/
+BOOST_AUTO_TEST_CASE(KDEMainInvalidMCEntryCoef)
+{
+  arma::mat reference = arma::randu<arma::mat>(1, 10);
+  arma::mat query = arma::randu<arma::mat>(1, 5);
+
+  // Main params.
+  SetInputParam("reference", reference);
+  SetInputParam("query", query);
+  SetInputParam("kernel", std::string("gaussian"));
+  SetInputParam("monte_carlo", true);
+
+  Log::Fatal.ignoreInput = true;
+  // Invalid under 1.
+  SetInputParam("mc_entry_coef", 0.5);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Valid greater than 1.
+  SetInputParam("mc_entry_coef", 1.1);
+  BOOST_REQUIRE_NO_THROW(mlpackMain());
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensure we get an exception when an invalid Monte Carlo break coefficient
+  * is specified.
+ **/
+BOOST_AUTO_TEST_CASE(KDEMainInvalidMCBreakCoef)
+{
+  arma::mat reference = arma::randu<arma::mat>(1, 10);
+  arma::mat query = arma::randu<arma::mat>(1, 5);
+
+  // Main params.
+  SetInputParam("reference", reference);
+  SetInputParam("query", query);
+  SetInputParam("kernel", std::string("gaussian"));
+  SetInputParam("monte_carlo", true);
+
+  Log::Fatal.ignoreInput = true;
+  // Invalid under 0.
+  SetInputParam("mc_break_coef", -0.5);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  // Valid between 0 and 1.
+  SetInputParam("mc_break_coef", 0.3);
+  BOOST_REQUIRE_NO_THROW(mlpackMain());
+
+  // Invalid greater than 1.
+  SetInputParam("mc_break_coef", 1.1);
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
```

Comment:
Since the Monte Carlo search has a random parameter, it may be worth running twice with `--monte_carlo` specified to ensure that the results are at least a little bit different (due to different random sampling).  This will help test that `--monte_carlo` is actually doing something. :)